tsl/src/chunk_copy.c

```diff
@@ -418,6 +435,71 @@ chunk_copy_stage_create_replication_slot(ChunkCopy *cc)
 	ts_dist_cmd_run_on_data_nodes(cmd, list_make1(NameStr(cc->fd.source_node_name)), true);
 }
 
+static void
+chunk_copy_stage_create_replication_slot_cleanup(ChunkCopy *cc)
+{
+	char *cmd;
+	DistCmdResult *dist_res;
+	PGresult *res;
+
+	/* Check if the slot exists on the source data node */
+	cmd = psprintf("SELECT 1 FROM pg_catalog.pg_replication_slots WHERE slot_name = '%s'",
+				   NameStr(cc->fd.operation_id));
+	dist_res =
+		ts_dist_cmd_invoke_on_data_nodes(cmd, list_make1(NameStr(cc->fd.source_node_name)), true);
+	res = ts_dist_cmd_get_result_by_node_name(dist_res, NameStr(cc->fd.source_node_name));
+
+	if (PQresultStatus(res) != PGRES_TUPLES_OK)
+		ereport(ERROR,
+				(errcode(ERRCODE_CONNECTION_EXCEPTION), errmsg("%s", PQresultErrorMessage(res))));
+
+	/* Drop replication slot on the source data node only if it exists */
+	if (PQntuples(res) != 0)
```

Comment:
I believe it should be impossible to have slots with the same name. I am thinking about the situation, when drop slot executed correctly inside `BEGIN/COMMIT` block but it still shows in that view, because the slot drop procedure is async, if that makes sense

Reply:
The cleanup typically as of now is done much later. So if the replication slot is around, it's around and not due to async cleanup.